src/business/dal/api/api.connection.ts

```diff
@@ -0,0 +1,48 @@
+import axios, { AxiosInstance } from 'axios';
+
+export default class Api {
+  private static _instance: Api;
+  private _database?: AxiosInstance;
+  private _apiURL?: string;
+
+  private constructor() {
+  }
+
+  public static getInstance(): Api {
+    if (Api._instance === undefined) {
+      Api._instance = new Api();
+    }
+
+    return Api._instance;
+  }
+
+  public get apiURL(): string | undefined {
+    return this._apiURL;
+  }
+
+  public initialize(apiURL: string): void {
+    this._apiURL = apiURL;
+
+    try {
+      this._database = axios.create({
+        baseURL: apiURL,
+      });
+    } catch (e) {
+      this._database = undefined;
+
+      throw e;
+    }
+  }
+
+  public isInitialized(): boolean {
+    return this._database !== undefined;
+  }
+
+  public get db(): AxiosInstance {
```

Comment:
### 🤔 Question: why do u need this method?

I think this class should be the only owner of this `_database` instanase. All who want to use it - have to do it throw this class interface.

Reply:
I don't understand what I should change :)
Could you describe more?